packages/chakra-components/src/MultiSelect/useComboBox.js

```diff
@@ -0,0 +1,581 @@
+import { useEffect, useRef, useCallback, useReducer } from "react";
+import debounce from "lodash.debounce";
+import { useForkRef } from "@chakra-ui/core/dist/utils";
+import { useConstant } from "./useConstant";
+
+const defaultFilter = (options, input) => {
+  if (input) {
+    return options.filter(option =>
+      option.label.toLowerCase().includes(input.toLowerCase()),
+    );
+  } else {
+    return options;
+  }
+};
+
+export const useComboBox = ({
+  options,
+  value: initialValues,
+  onChange,
+  filteredBy = defaultFilter,
+  isMulti,
+  isListBox,
+  isAsync,
+}) => {
+  // Refs
+  const listRef = useRef(null);
+  const inputRef = useRef(null);
+  const multiSelectRef = useRef(null);
+  const popperRef = useRef(null);
+
+  /**
+   * Check if the provided initial value/values are valid against the given options.
+   */
+  const isValuesValid = (values, options) => {
+    if (!isMulti) {
+      return options.some(option => option.value === values);
+    } else {
+      let isValid = true;
+
+      /**
+       * Used forEach to break the loop when it is not valid.
+       */
+      for (let i = 0; i < values.length; i++) {
+        if (!options.some(option => option.value === values[i])) {
+          isValid = false;
+          break;
+        }
+      }
```

Comment:
```suggestion
      filter(values, eachValue => options.some(option => option.value === eachValue)).length > 0
```

Or even you can use findIndex

Reply:
Fixed this.
![image](https://user-images.githubusercontent.com/39694575/82293254-ebb1dd80-99c9-11ea-97f1-51ad18b61e67.png)